src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs

```diff
@@ -200,10 +200,9 @@ private static void WriteConstructor(CodeWriter writer, ObjectType schema)
                     foreach (var initializer in constructor.Initializers)
                     {
                         writer.Append($"{initializer.Property.Declaration.Name} = ")
-                            .WriteConversion(initializer.Value.Type, initializer.Property.Declaration.Type, w=>w.WriteReferenceOrConstant(initializer.Value));
+                            .WriteConversion(initializer.Value.Type, initializer.Property.Declaration.Type, w => w.WriteReferenceOrConstant(initializer.Value));
 
-                        if (initializer.DefaultValue != null &&
-                            !initializer.Value.Type.IsValueType)
+                        if (initializer.DefaultValue != null && (!initializer.Value.Type.IsValueType || TypeFactory.IsStruct(initializer.Value.Type)))
```

Comment:
Would `initializer.Value.Type.IsNullable` work here? It's a bit more descriptive of what's actually happening.

Reply:
Nope, `initializer.DefaultValue != null && initializer.Value.Type.IsNullable` is not working here. There are many cases where there is a default value but it's not nullable. 

Example:
1. `!initializer.Value.Type.IsValueTyp` condition is needed for cases like: https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestServerProjects/body-complex/Generated/Models/Shark.cs#L36 (`fishtype` is not nullable here)

2. And `TypeFactory.IsStruct(initializer.Value.Type)` condition is needed for: https://github.com/ShivangiReja/autorest.csharp/blob/Generate_Content.json/test/TestServerProjects/constants/Generated/Models/ModelAsStringRequiredTwoValueDefault.cs#L17

Reply:
You are right about the first, the second would be covered by `initializer.Value.Type.IsNullable`.

What about?
```
  !initializer.Value.Type.IsValueType ||
   initializer.Value.Type.IsNullable
```
?

Reply:
Yes, it worked!